src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -1716,5 +1684,85 @@ private static bool ConvertExceptionMessageFromStringToMessageObject(JObject exc
 
             return modifiedNotification;
         }
+
+        private delegate void ActionOnJObject(JObject jObject);
+
+        private static void ApplyActionOnNodeIfPathExists(string[] possibleNodePaths, JObject rootNode, ActionOnJObject action)
+        {
+            var nodesToUpdate = FindAllNodes(possibleNodePaths, rootNode);            
+
+            foreach (JObject node in nodesToUpdate)
+            {
+                action(node);
+            }
+
+        }
+
+        private static List<JObject> FindAllNodes(string[] fullPaths, JObject rootlog)
+        {
+            var nodes = new List<JObject>();
+
+            foreach (string path in fullPaths)
+            {
+                nodes.AddRange(GetAllNodesAtPath(path, rootlog));
+            }
+
+            return nodes;
+        }
+
+        private static List<JObject> GetAllNodesAtPath(string fullPath, JObject rootLog)
```

Comment:
>GetAllNodesAtPath [](start = 37, length = 17)

one drawback - this algorithm will not handle recursive object references. (e.g. innerExceptions > exception) Perhaps we can introduce a special symbol for recursive references so tht we can keep looping recursively till all objects are found? i am thinking about it while this version is being reviewed.

Reply:
Write a test that thoroughly breaks our current approach due to this limitation, then mark it [Skip] and open a tracking to address the problem.

---
In reply to: [265813535](https://github.com/Microsoft/sarif-sdk/pull/1349#discussion_r265813535) [](ancestors = 265813535)